### PR TITLE
feat(ME-2223): upload recordings on interval

### PR DESCRIPTION
**Pull request by btoonk**
# Description

upload recordings on interval, now it's set to 30s or size is over 10MB.
Fixes # (ME-2223)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested in staging.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: internal/sqlauthproxy/recording.go [hunk 1]
```diff
@@ -39,65 +47,90 @@ func newRecording(logger *zap.Logger, sessionKey string, api border0.Border0API)
...
+		for {
+			select {
+			case message, open := <-messageChan:
+				if !open {
 					return
 				}
 
-				if err := r.zipWriter.Close(); err != nil {
-					r.logger.Error("failed to close session log file", zap.Error(err))
+				logJson, _ := json.Marshal(message)
```
Comment (adrianosela) [inline internal/sqlauthproxy/recording.go:74]: Catch err

File: internal/sqlauthproxy/recording.go [hunk 2]
```diff
@@ -39,65 +47,90 @@ func newRecording(logger *zap.Logger, sessionKey string, api border0.Border0API)
...
+		return fmt.Errorf("failed to close session log file: %s", err)
+	}
+
+	uploadBuffer := make([]byte, r.buf.Len())
+	copy(uploadBuffer, r.buf.Bytes())
+	r.buf.Reset()
+	r.zipWriter = gzip.NewWriter(&r.buf)
+
+	go func(uploadBuffer []byte) {
 		if err := r.api.UploadRecording(uploadBuffer, r.sessionKey, r.recordingID.String()); err != nil {
```
Comment (adrianosela) [inline internal/sqlauthproxy/recording.go:129]: Does this need a lock to guarantee order? Or does order not matter? (If one upload takes longer than the ticker time for example)

File: internal/sqlauthproxy/recording.go (hunk 1 above)
Comment (btoonk) [reply internal/sqlauthproxy/recording.go]: done

File: internal/sqlauthproxy/recording.go (hunk 2 above)
Comment (btoonk) [reply internal/sqlauthproxy/recording.go:140]: well maybe, but I think we will have other issues if this happens. But just to be sure I added a lock. Nice catch!

**Pull request merged**